kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ConditionType.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+public enum ConditionType {
+    Ready("Ready"),
+    Accepted("Accepted");
```

Comment:
Not unreasonable, but this is still in flux at the moment. I'm intending on following the Gateway API which has quite particular semantics about conditions, which differ between the different APIs (e.g. gateway and route have different conditions). 

They don't use the `Ready` type (which is what the Kubernetes API conventions recommend). I've kept ready for now for the `KafkaProxy.status.condition`. Instead they use `Accepted` and `Programmed`, which each have different allowed `reasons` (and the reaons you can use depend on the status too, because otherwise it would be too simple). See, for example https://github.com/kubernetes-sigs/gateway-api/blob/e16672035c3cdf1acf0143c36cee0425487f2734/apis/v1/shared_types.go#L290

I don't want to make this too elaborate right now though because my next move will be to factor out a `KafkaRoute`. At that point I'll be looking at what abstractions are possible to avoid code repetition while also providing some guarantees that we're building allowed combinations of CR, type, status and reason.

This is also made painful because of the class naming strategy the Fabric8 CRD-to-Java converter uses. Because it consumes a CRD, and because the Kube rules for CRDs say "`$ref` is not allowed" all the schemas are effectively "anonymous types". The converter has to honour this pattern, which means you end up with two distinct Java classes for identical structures which are simply used in different places in a single CRD. In this case we want to have a common Condition concept, but we end up with two Java classes (both called Condition, but in different packages).

So to be able to do nice abstractions about conditions we're likely to want to address that. Probably that means ditching the Fabric8 CRD-to-Java converter and using jsonschema-2-pojo to generate the Java classes. That would allow to use named schemas (the `$ref` and `definitions` keywords in OpenAPI-speak), but at the cost of needing to write a small tool which takes a JSON schema, recursively resolves the `$ref` and generates a `CustomResourceDefinition` around it. I'm holding off on this for now until it becomes too painful not to do it.